notary-commons/src/main/kotlin/sidechain/iroha/IrohaChainListener.kt

```diff
@@ -1,51 +1,130 @@
 package sidechain.iroha
 
+import config.RMQConfig
 import com.github.kittinunf.result.Result
 import com.github.kittinunf.result.map
-import io.reactivex.Observable
+import com.rabbitmq.client.*
 import jp.co.soramitsu.iroha.java.IrohaAPI
 import model.IrohaCredential
 import mu.KLogging
 import sidechain.ChainListener
 import sidechain.iroha.util.ModelUtil
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import kotlin.test.assertNotNull
+
 
 /**
  * Dummy implementation of [ChainListener] with effective dependencies
  */
 class IrohaChainListener(
```

Comment:
I strongly recommend splitting this listener into two implementations: plain Iroha block listener(like it was before) and reliable Iroha block listener.

Reply:
I will remove plain listener later